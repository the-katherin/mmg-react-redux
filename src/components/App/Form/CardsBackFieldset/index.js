import './CardsBackFieldset.css';

export { default } from './CardsBackFieldset';