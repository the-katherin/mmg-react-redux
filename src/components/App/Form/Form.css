.form {
  display: grid;
  justify-content: center;
  grid-row-gap: 10px; 
  padding: 30px;
  background: rgba(0, 120, 201, 0.7);
  border-radius: 10px;
  border: 1px solid white;
}

.form__submit-btn {
  justify-self: center;
}
