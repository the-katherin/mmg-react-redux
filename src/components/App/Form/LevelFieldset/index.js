import './LevelFieldset.css';

export { default } from './LevelFieldset';