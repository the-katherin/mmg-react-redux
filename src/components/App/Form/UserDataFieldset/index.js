import './UserDataFieldset.css';

export { default } from './UserDataFieldset';