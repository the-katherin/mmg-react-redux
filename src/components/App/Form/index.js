import './Form.css';

export { default } from './Form';
