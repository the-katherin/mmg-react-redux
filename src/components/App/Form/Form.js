import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import LevelFieldset from "./LevelFieldset";
import UserDataFieldset from "./UserDataFieldset";
import CardsBackFieldset from "./CardsBackFieldset";
import { getUserChoiceFromForm, getUserDataFromForm } from '../../../actions'

export class Form extends Component {

    constructor(props) {
        super(props);

        this.state = {
            redirect: false
        }

    }

    submitForm = (e) => {
        e.preventDefault();
        let form = e.target;
        let userName = form.userData.elements[0].value;
        let userLastName = form.userData.elements[1].value;
        let userEmail = form.userData.elements[2].value;;
        let level = form.level.value;
        let cardsBack = form.cardsBack.value;

        this.props.getUserChoiceFromForm(level, cardsBack);
        this.props.getUserDataFromForm(userName, userLastName, userEmail);
        this.setState({ redirect: true });

    }


    render() {
        const { redirect } = this.state;

        if (redirect) {
            return <Redirect to='/game' />;
        }


        return (
            <form className='form' name='userSelectionForm' onSubmit={this.submitForm} >

                <UserDataFieldset />
                <LevelFieldset />
                <CardsBackFieldset />

                <button className='form__submit-btn' type='submit' >Let's play</button>

            </form>
        );
    }
}

Form.propTypes = {
    getUserDataFromForm: PropTypes.func.isRequired,
    getUserChoiceFromForm: PropTypes.func.isRequired,
}

export default connect(
    null,
    {
        getUserDataFromForm,
        getUserChoiceFromForm
    }
)(Form);