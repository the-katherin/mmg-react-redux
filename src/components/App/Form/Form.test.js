import React from "react";
import Form from "./Form";

import configureStore from 'redux-mock-store';

const initialState = {
    getUserDataFromForm: jest.fn(),
    getUserChoiceFromForm: jest.fn(),
};

const mockStore = configureStore();
let wrapper;
let store;

describe("Form", () => {
    beforeEach(() => {
        store = mockStore(initialState)
        wrapper = mount(<Form store={store} />)
    });

    test("match snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('submit button exists', () => {
        const button = wrapper.find('button');
        expect(button).toBeDefined();
    });

});
