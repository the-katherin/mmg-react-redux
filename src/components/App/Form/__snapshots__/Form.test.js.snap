// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form match snapshot 1`] = `
<Connect(Form)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Form
    getUserChoiceFromForm={[Function]}
    getUserDataFromForm={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <form
      className="form"
      name="userSelectionForm"
      onSubmit={[Function]}
    >
      <UserDataFieldset>
        <fieldset
          className="user-data-fieldset"
          name="userData"
        >
          <legend>
            Fill the form below
          </legend>
          <div
            className="user-data-fieldset__inputs-grid"
          >
            <div>
              <input
                id="userName"
                pattern="[A-Za-zА-Яа-яЁё]{1,15}"
                placeholder="Enter your first name"
                required={true}
                type="text"
              />
            </div>
            <div>
              <input
                id="userLastName"
                pattern="[A-Za-zА-Яа-яЁё]{1,15}"
                placeholder="Enter your last name"
                required={true}
                type="text"
              />
            </div>
            <div>
              <input
                id="email"
                placeholder="Enter your email"
                required={true}
                type="email"
              />
            </div>
          </div>
        </fieldset>
      </UserDataFieldset>
      <LevelFieldset>
        <fieldset
          className="level-fieldset"
          name="level"
        >
          <legend>
            Choose level
          </legend>
          <div
            className="level-fieldset__radios"
          >
            <div>
              <input
                id="easy"
                name="level"
                required={true}
                type="radio"
                value="3"
              />
              <label
                htmlFor="easy"
              >
                Easy
              </label>
            </div>
            <div>
              <input
                id="medium"
                name="level"
                type="radio"
                value="6"
              />
              <label
                htmlFor="medium"
              >
                Medium
              </label>
            </div>
            <div>
              <input
                id="professional"
                name="level"
                type="radio"
                value="12"
              />
              <label
                htmlFor="professional"
              >
                Professional
              </label>
            </div>
          </div>
        </fieldset>
      </LevelFieldset>
      <CardsBackFieldset>
        <fieldset
          className="cards-back-fieldset"
          name="cardsBack"
        >
          <legend>
            Choose cards' back
          </legend>
          <div
            className="cards-back-fieldset__radios"
          >
            <div>
              <input
                id="cardsback1"
                name="cardsBack"
                required={true}
                type="radio"
                value="1"
              />
              <label
                htmlFor="cardsback1"
              >
                <img
                  alt="first cards' back"
                  src="cardsback1.jpg"
                />
              </label>
            </div>
            <div>
              <input
                id="cardsback2"
                name="cardsBack"
                type="radio"
                value="2"
              />
              <label
                htmlFor="cardsback2"
              >
                <img
                  alt="sec cards' back"
                  src="cardsback2.jpg"
                />
              </label>
            </div>
            <div>
              <input
                id="cardsback3"
                name="cardsBack"
                type="radio"
                value="3"
              />
              <label
                htmlFor="cardsback3"
              >
                <img
                  alt="third cards' back"
                  src="cardsback3.jpg"
                />
              </label>
            </div>
          </div>
        </fieldset>
      </CardsBackFieldset>
      <button
        className="form__submit-btn"
        type="submit"
      >
        Let's play
      </button>
    </form>
  </Form>
</Connect(Form)>
`;
